Assets/HoloToolkit/Common/Scripts/Singleton.cs

```diff
@@ -62,11 +72,48 @@ public static bool IsInitialized
             }
         }
 
+        public static bool ConfirmInitialized()
+        {
+            T access = Instance;
+            return IsInitialized;
+        }
+
+        [SerializeField]
+        private bool DontDestroyParentRootOnLoad = true;
+
+        private object initializedLock = new object();
+
+        protected void Initialize()
+        {
+            lock (initializedLock)
+            {
+                if (!IsInitialized)
+                {
+                    instance = (T)this;
+                    InitializeInternal();
+                    if (DontDestroyParentRootOnLoad)
+                    {
+                        instance.gameObject.GetParentRoot().DontDestroyOnLoad();;
```

Comment:
We should probably use `DontDestroyOnLoad(instance.transform.root);`

Also, remove extra `;`